src/NLog/LogFactory.cs

```diff
@@ -865,10 +872,51 @@ protected virtual void Dispose(bool disposing)
 #endif
         }
 
-        private static IEnumerable<string> GetCandidateConfigFileNames()
+        /// <summary>
+        /// Get file paths (including filename) for the possible NLog config files. 
+        /// </summary>
+        /// <returns>The filepaths to the possible config file</returns>
+        public IEnumerable<string> GetCandidateConfigFilePaths()
+        {
+            if (candidateConfigFilePaths != null)
+            {
+                return candidateConfigFilePaths.AsReadOnly();
+            }
+
+            return GetDefaultCandidateConfigFilePaths();
+        }
+
+        /// <summary>
+        /// Overwrite the paths (including filename) for the possible NLog config files.
+        /// </summary>
+        /// <param name="filePaths">The filepaths to the possible config file</param>
+        public void SetCandidateConfigFilePaths(IEnumerable<string> filePaths)
+        {
+            candidateConfigFilePaths = new List<string>();
+
+            if (filePaths != null)
+            {
+                foreach (var filePath in filePaths)
+                {
+                    candidateConfigFilePaths.Add(filePath);
```

Comment:
Thanks, it was first a `IList` but with the move to `List`, AddRange is indeed better